e3sm_diags/driver/zonal_mean_2d_stratosphere_driver.py

```diff
@@ -12,7 +11,9 @@ def create_metrics(ref, test, ref_regrid, test_regrid, diff):
     return base_create_metrics(ref, test, ref_regrid, test_regrid, diff)
 
 
-def run_diag(parameter):
+def run_diag(
+    parameter: ZonalMean2dStratosphereParameter,
+) -> ZonalMean2dParameter:
```

Comment:
@tomvothecoder I'm not exactly sure why using `coreParameter` here triggers error "NameError: name 'CoreParameter' is not defined" at the first place and  then `mypy` insist to expect  ZonalMean2dParameter. But things run okay after my changes.

Reply:
Looks good with this change.

For your error:
---
You need to add `from __future__ import annotations` to reference the `CoreParameter` module without quotes, otherwise you get that `NameError`.

Without `import annotations`:
```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from e3sm_diags.parameter.core_parameter import CoreParameter

def run_diag(
	parameter: ZonalMean2dStratsophereParameter
) -> "CoreParameter"
```

With `import annotations`:
```
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from e3sm_diags.parameter.core_parameter import CoreParameter

def run_diag(
	parameter: ZonalMean2dStratsophereParameter
) -> CoreParameter
```
This is a "[postponed evaluation of annotations](https://peps.python.org/pep-0563/)", which I believe means evalutation only happens through a typechecker like `mypy` and not at runtime. It is also cleaner without quotes in my opinion.

Reply:
Thanks! Good to learn this!